### PR TITLE
FEATURE: LX-160, LX-163 Disable selected date filters and telemetry

### PR DESCRIPTION
**Pull request by NestorEncinas**
- Disable selected date filters with proper explanation tooltip.

JIRA: LX-160

- Telemetry

JIRA: LX-163

---

> [!IMPORTANT]
> Please, **don't forget to run `rush change`** for the commits that introduce **new features** 🙏

---

Refer to [documentation](https://github.com/gooddata/gooddata-ui-sdk/blob/master/dev_docs/continuous_integration.md) to see how to run checks and tests in the pull request. This is the list of the most used commands:

```
extended test - backstop
```

```
// Tiger platform
extended test - tiger-cypress - integrated
extended test - tiger-cypress - isolated
extended test - tiger-cypress - record
```

```
// Bear platform
extended test - cypress - integrated
extended test - cypress - isolated
extended test - cypress - record
```

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18399/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/20037/ : SUCCESS in 8m 09s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18400/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/20038/ : SUCCESS in 8m 23s

**Comment by NestorEncinas** [comment]
extended test - tiger-cypress - isolated

**Comment by NestorEncinas** [comment]
extended test - cypress - isolated

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18403/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/2074/ : SUCCESS in 11m 08s

**Comment by NestorEncinas** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18406/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/20042/ : SUCCESS in 8m 02s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18407/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/20043/ : SUCCESS in 8m 06s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18409/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/20045/ : SUCCESS in 8m 03s

**Comment by NestorEncinas** [comment]
extended test - tiger-cypress - isolated

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18411/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/2076/ : SUCCESS in 11m 51s

File: libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/limitValues/shared/limitingItemsHook.ts [hunk 1]
```diff
@@ -390,3 +401,28 @@ const parseCommonDateFilter = (
         isCommonDate: true,
     };
 };
+
+export const useCommonDateFilterTitle = (intl: IntlShape): string => {
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/limitValues/shared/limitingItemsHook.ts:405]: seems to be quite generic hook, is this file the right location for it?

File: libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/limitValues/shared/limitingItemsHook.ts [hunk 2]
```diff
@@ -390,3 +401,28 @@ const parseCommonDateFilter = (
...
+    return useMemo(() => {
+        const dateFilter = dependentDateFilters.find((filter) =>
+            areObjRefsEqual(filter.dataSet, (item as IDashboardDependentDateFilter).dataSet),
+        );
+
+        const dateDataSet = dateFilter?.dataSet ? dateDataSetsMap.get(dateFilter?.dataSet) : "";
+        const dataSetTitle = dateDataSet ? dateDataSet.dataSet.title : "";
+
+        return getDatasetTitle(dateFilter?.dataSet, dataSetTitle, filterConfigByDimension);
+    }, [dependentDateFilters, item, filterConfigByDimension]);
```
[the remark this answers is not in the document]
Comment (NestorEncinas) [reply libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/limitValues/shared/limitingItemsHook.ts:427]: Good catch, thanks.

File: libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/limitValues/shared/limitingItemsHook.ts [hunk 3]
```diff
@@ -390,3 +401,28 @@ const parseCommonDateFilter = (
...
+export const useDependentDateFilterTitle = (
+    item: ValuesLimitingItem,
+    dependentDateFilters: IDashboardDependentDateFilter[],
+): string => {
+    const dateDataSetsMap = useDashboardSelector(selectAllCatalogDateDatasetsMap);
+    const filterConfigByDimension = useDashboardSelector(selectDateFilterConfigsOverrides);
+
+    return useMemo(() => {
+        const dateFilter = dependentDateFilters.find((filter) =>
+            areObjRefsEqual(filter.dataSet, (item as IDashboardDependentDateFilter).dataSet),
```
[the remark this answers is not in the document]
Comment (NestorEncinas) [reply libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/limitValues/shared/limitingItemsHook.ts:420]: Updated.

File: libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/limitValues/shared/limitingItemsHook.ts (hunk 1 above)
Comment (NestorEncinas) [reply libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/limitValues/shared/limitingItemsHook.ts:405]: Moved to `DateFilter` folder structure.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18416/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/20052/ : SUCCESS in 8m 03s

**Review by ivan-nejezchleb — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR4810

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/4082/ : SUCCESS in 7m 45s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/1961/ : SUCCESS in 1m 04s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/20058/ : SUCCESS in 8m 02s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/1753/ : SUCCESS in 12m 01s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/1738/ : SUCCESS in 10m 04s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/5256/ : SUCCESS in 8s

**Pull request merged**